sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ImplementationBridgeHelpers.java

```diff
@@ -106,13 +109,16 @@ private PartitionKeyHelper() {}
 
         public static void setPartitionKeyAccessor(final PartitionKeyAccessor newAccessor) {
             if (!accessor.compareAndSet(null, newAccessor)) {
-                logger.warn("PartitionKeyAccessor already initialized!");
+                logger.debug("PartitionKeyAccessor already initialized!");
+            } else {
+                logger.info("Setting PartitionKeyAccessor...");
+                partitionKeyClassLoaded.set(true);
             }
-            partitionKeyClassLoaded.set(true);
         }
 
         public static PartitionKeyAccessor getPartitionKeyAccessor() {
-            if (partitionKeyClassLoaded.compareAndSet(false, true)) {
+            if (!partitionKeyClassLoaded.get()) {
+                logger.debug("Initializing PartitionKeyAccessor...");
```

Comment:
Are we changing the logger level to `debug` because we expect this race condition to be hit more often?

Reply:
Before there was no race condition (for the inner part of the if-block) - the line would have been logged exactly once. Now when multiple threads try to get the accessor concurrently multiple of them would log (until the accessor has actually been initialized by a single thread). I changed the log level because we might see multiple of these logs - up to number of cores. I am ok with still using info if you think it is needed (I think the two info logs should be sufficient - but given that we have spent 3 PRs on this open for changing to info)